jujugui/static/gui/src/app/components/charmbrowser/charmbrowser.js

```diff
@@ -54,34 +79,41 @@ YUI.add('charmbrowser-component', function() {
       @return {Object} A generated state object which can be passed to setState.
     */
     generateState: function(nextProps) {
-      var metadata = nextProps.appState.sectionC.metadata;
-      var utils = this.props.utils;
-      var state = {
-        activeComponent: metadata.activeComponent
+      return {
+        activeComponent: nextProps.appState.sectionC.metadata.activeComponent
       };
-      switch (state.activeComponent) {
+    },
+
+    /**
+      Generate the content based on the state.
+
+      @method _generateContent
+      @return {Object} The child components for the content.
+    */
+    _generateContent: function() {
```

Comment:
I had to move the content generation out of state so that it can be modified on state updates.